### ISSUE TITLE
Picker should migrate to Select-Only Combobox Example from WAI-ARIA Authoring Practices Guide 

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#4091]
# 🐛 Bug Report 

Picker implements the [(Deprecated) Collapsible Dropdown Listbox Example](https://w3c.github.io/aria-practices/examples/listbox/listbox-collapsible.html) from an outdated version of the WAI-ARIA 1.2 Authoring Practices Guide. We should update Picker to follow the latest recommendation, [Select-Only Combobox Example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/).

## 🤔 Expected Behavior

Picker should be announced as a combo box and with its text content as its value, which should reduce the need to specify `aria-labelledby` for both the label and the value content.

## 😯 Current Behavior

Picker announces as a dropdown menu button.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | [@adobe/react-spectrum@3.24.1](https://github.com/adobe/react-spectrum/releases/tag/%40adobe%2Freact-spectrum%403.24.1)
| Browser          | Chrome Version 110.0.5481.100 (Official Build) (arm64)
| Operating System | macOS Ventura 13.2.1 (22D68)

## 🧢 Your Company/Team

Adobe/Accessibility

## 🕷 Tracking Issue (optional)

A11Y-5854

**Comment by devongovett** [comment]
What's the advantage of doing this? I tested the select-only combobox example, and VoiceOver on macOS doesn't announce the options at all when navigating with the arrow keys.

**Comment by majornista** [comment]
> What's the advantage of doing this? I tested the select-only combobox example, and VoiceOver on macOS doesn't announce the options at all when navigating with the arrow keys.

I'm not sure we would have the same issue with VoiceOver and Safari on macOS, which is an open issue against WebKit (https://bugs.webkit.org/show_bug.cgi?id=231724), because with Picker, we focus items in the listbox directly rather than use `aria-activedescendant`. With Chrome on macOS, `aria-activedescendant` is working correctly. 

Advantages:
1. We would be following the latest WAI-ARIA APG recommendation. To find [(Deprecated) Collapsible Dropdown Listbox Example](https://w3c.github.io/aria-practices/examples/listbox/listbox-collapsible.html), I had to dig back through old versions of the APG.  
2. Simplifies labelling, because we would no longer need to concatenate the IDREFs for the label and the value. 
3. Screen reader announcement of value and label would be more consistent with platform and AT conventions, adding commas where appropriate, etc.

**Comment by devongovett** [comment]
Yeah I'm just concerned that it is a pretty big change for not much benefit overall given we already have a working implementation and there aren't really any major problems with it that this would solve. We would need to invest a lot of time into testing and working around screen reader/browser issues again, which we already did for the last implementation. We also had a ton of problems with announcements in ComboBox (which can be seen in the APG example itself) so it would be unfortunate to need to solve all of that again.

**Comment by LFDanLu** [comment]
Closing due to what seems to be limited upside, but we may revisit in the future, particularly with upcoming autocomplete work that will result in a refactor of the combobox hooks
